Fix for #854 - cleaner rosout log rotation

mgrrx opened the pull request:
This PR fixes #854 by rotating the log files as usual in Linux to guarantee the ordering of log files (https://en.wikipedia.org/wiki/Log_rotation).

peci1 committed:
Allowed saving timestamp-less messages to Cache (using rospy.Time.now() as their timestamp).

Also added a convenience `getLast` method for retrieving the last message in the cache.

wkentaro committed:
Get multiple topics for rostopic hz

wkentaro committed:
Abstract class to implement connection based transport

rhaschke committed:
roslaunch: python evaluation of expressions

- evaluation of expressions of the form $(eval ...)
- functions env(), optenv(), anon(), find(), arg() map onto corresponding $-substitutions
- implicit access to args using their name, i.e. name instead of arg('name')

wkentaro committed:
rostopic echo: Show stat when --noarr/nostr

eric-wieser committed:
Implement `__contains__`, which is a better spelling of `has`

gerkey committed:
Enabled tests and then fixed them, with much help from @dhood

dirk-thomas committed:
reenabled roswtf tests

dirk-thomas committed:
allow roswtf test to pass without rosdep being initialized (which is the case on the farm)

dirk-thomas committed:
add missing parameter to AdvertiseOptions::createAdvertiseOptions (#733)

dirk-thomas committed:
make LogAppender and Token destructor virtual (#729)

dirk-thomas committed:
change return value of param() to bool (#753)

dirk-thomas committed:
improve TopicManager::instance (#770)

eric-wieser committed:
Promote the result of read_messages to a namedtuple

dirk-thomas committed:
Merge pull request #773 from ros/log_appender_virtual_destructor

make LogAppender destructor virtual

dirk-thomas committed:
Merge pull request #774 from ros/missing_parameter

add missing parameter to AdvertiseOptions::createAdvertiseOptions

dirk-thomas committed:
Merge pull request #775 from ros/add_return_value_to_param

change return value of param() to bool

dirk-thomas committed:
Merge pull request #776 from ros/improve_topicmanager_instance

improve TopicManager::instance

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.12.0

gerkey committed:
switch to using std_msgs for tests

dhood committed:
Regression test for issue #736

codebot committed:
Use defusedxml in rosmaster

eric-wieser committed:
Remove duplicated function, use decorator

Requires ros/genpy#52 to be merged/released first

jspricke committed:
Fix confusing copyright messages/dates

Thanks to Wookey

dirk-thomas committed:
use directory specific compiler flags

gerkey committed:
Add missing inter-test dependency

gerkey committed:
Merge pull request #788 from ros/fix_test_rosbag_deps

Add missing inter-test dependency

dirk-thomas committed:
Merge pull request #782 from ros/rosmaster-defusedxml

Use defusedxml in rosmaster

dirk-thomas committed:
Merge pull request #785 from ros/directory_specific_compiler_flags

use directory specific compiler flags

dirk-thomas committed:
fix directory property

benureau committed:
Fix str conversion in encode_ros_handshake_header

The previous code was not properly filtering unicode/str
out of the header. Added the corresponding unit test.

dirk-thomas committed:
fix passing multiple args to add_rostest (fix #790)

dirk-thomas committed:
Merge pull request #791 from ros/fix_multiple_args_add_rostest

fix passing multiple args to add_rostest

dirk-thomas committed:
Merge pull request #792 from ros/fixstrconv2

fix header encoding in encode_ros_handshake_header

dirk-thomas committed:
Merge pull request #784 from ubi-agni/roslaunch-python-evaluation

python evaluation of expressions in roslaunch

rhaschke committed:
added math symbols and some standard data types for python evaluation

... and added appropriate unittests

dirk-thomas committed:
Merge pull request #793 from ubi-agni/roslaunch-math-expressions

added math symbols and some standard data types for python evaluation

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.12.1

dirk-thomas committed:
the log dir itself should not be symlinked as latest

codebot committed:
explicitly declare rosout message dependencies

dirk-thomas committed:
Merge pull request #799 from ros/add_rosgraph_msgs_dependency

explicitly declare rosout message dependencies

dirk-thomas committed:
update test to match new genpy behavior

dirk-thomas committed:
Merge pull request #803 from ros/update_test_to_new_genpy

update test to match new genpy behavior

dirk-thomas committed:
Merge pull request #780 from dhood/issue736

writing rosbags adds a newline

jasonimercer committed:
Check for pointer return from createOutgoing and fail requestTopic if it is NULL.

Commit:
Add a non verbose (quiet) mode for rosnode info

dirk-thomas committed:
Merge pull request #809 from spaghetti-/kinetic-devel

Add a non verbose (quiet) mode for rosnode info

jasonimercer committed:
Changed calling_in_this_thread restore mechanism in callOneCB from try-catch to RAII.

wkentaro committed:
Implement rospy.logXXX_throttle

Commit:
resolving naming conflicts when multiple test are added with arguments

Commit:
fix passing multiple args to roslaunch_add_file_check

Commit:
handle multiple environment arguments too

dirk-thomas committed:
Merge pull request #807 from jasonimercer/null-transport-check

Check for pointer return from createOutgoing and fail requestTopic if it is NULL.

dirk-thomas committed:
Merge pull request #811 from jasonimercer/do-not-catch-user-exception-callOneCB

Changed calling_in_this_thread restore mechanism in callOneCB from try-catch to RAII.

dirk-thomas committed:
Merge pull request #795 from ros/fix_latest_log_symlink

the log dir itself should not be symlinked as latest

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.12.2

Commit:
More detailed help string for rostopic echo -p

dirk-thomas committed:
remove extra semicolon

dirk-thomas committed:
Merge pull request #817 from ros/remove_extra_semicolon

remove extra semicolon

dirk-thomas committed:
Merge pull request #816 from spaghetti-/kinetic-devel

More detailed help string for rostopic echo -p

dirk-thomas committed:
improve debug message when queue is full (#818)

dirk-thomas committed:
Merge pull request #826 from ros/enable_tests3

Enabled tests and then fixed them

Commit:
Use logging Formatter, enabling printing exception info with exc_info=1

This recovers the funcitonality provided by the python logging library,
which includes exception information if the kwarg exc_info is provided
with a True value

Commit:
add_rostest_gtest does now add the created gtest-target as a dependeny to the created rostest

xqms committed:
rosbag_storage: make Bag constructor explicit

This catches common mistakes such as view.addQuery(bagFileName, ...)

Signed-off-by: Max Schwarz <max.schwarz@uni-bonn.de>

mikepurvis committed:
Drop strict-aliasing warnings from roslz4 file.

mikepurvis committed:
Fix to -Wno-strict-aliasing

dirk-thomas committed:
Merge pull request #839 from mikepurvis/patch-1

Drop strict-aliasing warnings from roslz4 file.

de-vri-es committed:
roslaunch: Fix <param command="..." /> for python3.

In python3, the standard output of the subprocess is of type bytes. That
causes the data to be stored as a binary xmlrpc node which ends up
base64 encoded when retrieving it later. Additionally, it crashed the
rosparam command line utility which could not deal with that.

This commit checks if the read command output is not of type `str` and
decodes it as UTF-8 in that case. This is a NOP in python2. We can't
check if the read output is of type bytes, because in python2 `bytes`
is just an alias for `str`.

dirk-thomas committed:
Merge pull request #840 from delftrobotics/roslaunch-param-command-python3

roslaunch: Fix <param command="..." /> for python3.

dirk-thomas committed:
fix compiler warnings in release mode

dirk-thomas committed:
set lz4_FOUND in order to continue using it with catkin_package(DEPENDS) (ros/catkin#813)

dirk-thomas committed:
Merge pull request #842 from ros/fix_warnings_in_release

fix compiler warnings in release mode

wkentaro committed:
Fix rostopic hz for multiple topics without no message

To fix below error

```
% rostopic hz /input_0 /input_1
subscribed to [/input_0]
subscribed to [/input_1]
no new messages
no new messages
topic      rate    min_delta   max_delta    std_dev    window
==============================================================
/input_0   9.997    0.0999      0.1002      8.891e-05   7
/input_1   0.9998   0.9995      1.001       0.0005633   7

topic      rate    min_delta   max_delta    std_dev    window
==============================================================
/input_0   9.998    0.0999      0.1002      7.342e-05   17
/input_1   0.9998   0.9995      1.001       0.0004887   17

topic      rate    min_delta   max_delta    std_dev    window
==============================================================
/input_0   9.999    0.09982     0.1002      7.967e-05   21
/input_1   0.9999   0.9995      1.001       0.0004702   21

no new messages
no new messages
no new messages
Traceback (most recent call last):
File "/home/wkentaro/ros/indigo/devel/bin/rostopic", line 6, in <module>
exec(fh.read())
File "<string>", line 35, in <module>
File "<string>", line 1977, in rostopicmain
File "<string>", line 1385, in _rostopic_cmd_hz
File "<string>", line 273, in _rostopic_hz
File "<string>", line 221, in print_hz
File "<string>", line 228, in _get_ascii_table
ValueError: max() arg is an empty sequence
```

dirk-thomas committed:
Merge pull request #843 from wkentaro/rostopic-hz-no-messages

Fix rostopic hz for multiple topics without no message

luke-gumbley committed:
Add 'Darwin' to unix-like platforms

Prevents issue calling socket.getaddrinfo with the machine name where this is not a known local hostname (e.g. where the machine name hasn't been added to the hosts file).

de-vri-es committed:
Fix `rosservice call` for python3.

dirk-thomas committed:
Merge pull request #847 from delftrobotics/rosservice-python3

Fix `rosservice call` for python3.

dirk-thomas committed:
Merge pull request #844 from jspricke/fix_years

Fix confusing copyright messages/dates

jonfink committed:
Increase reqeust_queue_size for xmlrpc server (#849)

Handles issues with many simultaneous connections to XmlRPC Server in
OSX 10.11

atiderko committed:
rospy: fixed reconnection on timeout and some other common errors

while roscpp reconnects on timeout or temporary 'No route to host'
errors the rospy does it not. So on connection problems longer than 3
min the connection between subscriber and publisher goes.
I added some exceptions if these occurs rospy does not close the socket
and reconnect now.

rospy: limited exponential backoff

since rospy reconnects on timeout now, the reconnects goes very fast in
times where it easier to restart the ros node then wait for a reconnect.
So I added a maximal backoff time.

rospy: added reconnection for topic connect while node start

on connection problems while node start the subscription to a publisher
will be stopped after 3 tries. Now the reconnection is stopped on
shutdown of the node.

dirk-thomas committed:
Merge pull request #830 from cwecht/rostest_dependencies_bin

 add_rostest_gtest does now add the created gtest-target as a depende…

dirk-thomas committed:
Merge pull request #835 from xqms/rosbag-explicit-constructor

rosbag_storage: make Bag constructor explicit

wkentaro committed:
[message_filters] Fast approximate time synchronization in message_filters (pure python) (#802)

* Fast approximate time synchronization in message_filters

* Don't break the api of message_filters Synchronizers

* Required changes to merge origin/kinetic-devel

dirk-thomas commented:
Please avoid any kind of unrelated changes and refactorings since it makes it more difficult to read and review the patch as well as more difficult to cherry-picking / backporting between different branches.

Please make future contributions against the latest branch (in this case kinetic-devel). I can cherry pick the patch in this case so no need to open a new PR for this.

Commit:
update rosbag filter progress meter to use raw uncompressed input size (#857)

130s committed:
[rostest] Add reusable node to test parameters.

wkentaro committed:
List message files only for specified package

As below

```
% rosmsg list --help
Usage: rosmsg list [package_name]

Options:
  -h, --help  show this help message and exit

% rosmsg list std_msgs
std_msgs/Bool
std_msgs/Byte
std_msgs/ByteMultiArray
std_msgs/Char
std_msgs/ColorRGBA
std_msgs/Duration
std_msgs/Empty
std_msgs/Float32
std_msgs/Float32MultiArray

% rosmsg list std_msg
WARNING: no msg is found for package [std_msg]

% rosmsg list
```

dirk-thomas committed:
Revert "List message files only for specified package"

This reverts commit 4192c864160f99ea87db03b85fa9f8ac79b5e070.

contradict committed:
Allow 64-bit sizes to be passed to robag max_size (#865)

contradict committed:
Record a maximum number of splits and then begin deleting old files (#866)

* Record a maximum number of splits and then begin deleting old files

* More helpful help string

* Correct indentation

rhaschke committed:
fix unknown dependency

rhaschke committed:
fix static destruction order

According to C++ standard (http://en.cppreference.com/w/cpp/utility/program/exit)
static variables and registered atexit handlers are destroyed/called in
reverse order of their occurence.

As the singletons TopicManager ... XMLRPCManager are created in
ros::start() after having ros::init() registering the atexit handler,
they will be destroyed before the call to the atexit handler, which
attempted to access them and call their shutdown methods again.

Use of global variables should be avoided in general, because their
destruction order is determined by library linking. Using singletons
the order of destruction can be determined by calling order.

dirk-thomas committed:
Merge pull request #869 from ubi-agni/kinetic-devel

fix unknown msg-generation dependency

wkentaro committed:
Fix rostopic echo for list of ROS message to fix #868

Fix #868

dirk-thomas committed:
fix order of init and publisher in example

dirk-thomas committed:
Merge pull request #873 from ros/fix_order_in_example

fix order of init and publisher in example

wkentaro committed:
add comment of using <rosparam> for a test node

add comment of using <rosparam> for a test node

wkentaro committed:
[rostopic] Show topic field type with rostopic type (#860)

As shown below:

```bash
% rostopic type /raw_image_bgr/image_color/foo
unknown topic type [/raw_image_bgr/image_color/foo]

% rostopic type /raw_image_bgr/image_color/header
sensor_msgs/Image header std_msgs/Header

% rostopic type /raw_image_bgr/image_color/header/frame_id
sensor_msgs/Image header/frame_id string

% rostopic type /raw_image_bgr/image_color
sensor_msgs/Image
```

wkentaro committed:
Fix wrong type in docstring for rospy.Timer

rospy.Time -> rospy.Duration for period.

wkentaro committed:
change rospy default rosconsole format for consistency with roscpp
add missing walltime to roscpp logging

dirk-thomas committed:
register nosetests only when testing is enabled (#880)

jliviero committed:
Added condition to skip empty ('') <include> paths (#882)

dirk-thomas committed:
Merge pull request #872 from wkentaro/fix-rostopic-echo-868

[rostopic] Fix rostopic echo for list of ROS message to fix #868

dirk-thomas committed:
Merge pull request #878 from wkentaro/rospy-timer-doc-typo

[rospy] Fix wrong type in docstring for rospy.Timer

wkentaro committed:
[rostest] Add test node if topic message is published at least once (#863)

Add test node if topic message is published at least once

dirk-thomas committed:
set default values for min_space and min_space_str

dirk-thomas committed:
Merge pull request #884 from ros/init_min_size

set default values for min_space and min_space_str

wkentaro committed:
Fix for no messages for one topic in two topics

Close https://github.com/ros/ros_comm/pull/712#issuecomment-244490918

k-okada committed:
support use_test_depends option for roslaunch-check

dirk-thomas committed:
Merge pull request #886 from wkentaro/rostopic-hz-multi

[rostopic] Fix for no messages for one topic in two topics

dirk-thomas committed:
fix rostopic hz to print no message when there are no message on any topic

dirk-thomas committed:
Merge pull request #888 from ros/fix_rostopic_hz_no_msgs_multiple_topics

fix rostopic hz to print no message when there are no message on any topic

dirk-thomas committed:
update roswtf tests for gennodejs

dirk-thomas committed:
update changelogs

jliviero committed:
Added missing arg to check_roslaunch_dir() definition (#890)

* Added missing arg to check_roslaunch_dir() definition

* Added use_test_depends arg to check_roslaunch_file() call

dirk-thomas committed:
Merge pull request #871 from ubi-agni/fix-static-destruction-order

fix static destruction order

dirk-thomas committed:
fix parameter name (fix #893)

furushchev committed:
throw exception if node name passed to init is empty (#894, #895)

dirk-thomas committed:
restore API compatibility of rostopic.ROSTopicHz

dirk-thomas committed:
Merge pull request #896 from ros/restore_api_compatibility_rostopic

restore API compatibility of rostopic.ROSTopicHz

mikepurvis committed:
Better naming for roslaunch check test XMLs. (#897)

* Better naming for roslaunch check test XMLs.

rhaschke committed:
fix multi-threaded spinning (#867)

* replace global spinmutex with SpinnerMonitor

Using a single recursive mutex disables to run several spinners in
parallel (started from different threads) - even if they operate on
different callback queues.

The SpinnerMonitor keeps a list of spinning callback queues, thus
making the monitoring local to callback queues.

* always start spinning, regardless of error

* making the error fatal

* fixup! replace global spinmutex with SpinnerMonitor

* activate unittest for spinners

- moved test/test_roscpp/test/test_spinners.cpp -> test/test_roscpp/test/src/spinners.cpp
- created rostest test/test_roscpp/test/launch/spinners.xml
- use thrown exception to evaluate error conditions

* throw a std::run_time exception when spinner couldn't be started

This correctly indicates the fatality of the error but allows for graceful quitting too.

* correctly count number of started multi-threaded spinners

fixes unittest Spinners.async

* addressed comments

* bug fix: it wasn't reassigned correctly

* removed redundant line of code

* turn sanity check into assertion

* turn error into warning

* rollback strict error checking for backwards compatibility

Previously, we could have the following situations:

1. `S1` (single-threaded spinner started in `thread 1`): will block `thread 1` until shutdown. Any further spinners in different threads were not allowed (with an error message).
2. `M1` (multi-threaded spinner started in `thread 1`): Further spinners started from _different_ threads were not allowed (with an error message).
3. `M1 ... M1 S1` (multi-threaded spinners started in `thread 1` and afterwards a single-threaded one started): This was accepted without any errors. But the new behavior is to reject `S1`!

Restrictions of case 1 + 2 are relaxed with this PR: Other spinners are allowed as long as the operate on a different queue. Thread doesn't matter.

The tricky part is case 3, which - although nonsense - was perfectly valid code before.
In order to maintain the old behavior, I need to remember, which thread the first M-spinner was started in, using the new variable `initial_tid`.

* allow spinning of a single-threaded spinner after some multi-threaded ones, as long as they are started from the same thread
* don't throw exceptions
* disabled corresponding unittests

* code simplification + improved comments

Allow multiple single-threaded spinners (in same thread)
and count their number.
Thus single-threaded and multi-threaded spinners are handled similarly.

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.12.3

dirk-thomas committed:
fix test type handling (#722)

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.12.4

dirk-thomas committed:
remove invalid export

jspricke committed:
Use boost::bind to bind the callback function

Example usage:
void callback(const boost::shared_ptr<M const>&)

message_filters::Subscriber<std_msgs::UInt32> sub(nh, "my_topic", 1);
sub.registerCallback(myCallback);

cf. http://wiki.ros.org/message_filters#Example_.28C.2B-.2B-.29

wkentaro committed:
[rostopic] Fix rostopic echo for non rosmsg field (#909)

* Fix rostopic echo for non rosmsg field

Close https://github.com/ros/ros_comm/issues/908

* Show stat for echoing non ROS message

* Fix field stat information with --nostr for string[]

* Add function of field value transformation

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.12.5

dirk-thomas committed:
add USE_TEST_DEPENDENCIES option to roslaunch_add_file_check()

Peter-Levine committed:
Add '#Include <vector>' to fix building on GCC-6

Don't rely on transitive header inclusion to declare std::vector as building with GCC-6 fails due to no '#Include <vector>' statement.

dirk-thomas committed:
Merge pull request #911 from Hodorgasm/kinetic-devel

Add '#Include <vector>' to fix building on GCC-6

wkentaro committed:
Fix typo of arg for _str_plot function

Fix #915

dirk-thomas committed:
Merge pull request #906 from jspricke/msg_filters_add_callback

Use boost::bind to bind the callback function

dirk-thomas committed:
Merge pull request #910 from ros/roslaunch_check_use_test_deps

add USE_TEST_DEPENDENCIES option to roslaunch_add_file_check()

dirk-thomas committed:
Merge pull request #916 from wkentaro/typo-str-plot-arg

[rostopic] Fix typo of arg for _str_plot function

dirk-thomas committed:
Merge pull request #783 from eric-wieser/patch-5

Remove duplicated function, use decorator

Commit:
Fix BagMigrationException in migrate_raw

Take the ._type using [0], because the arguments passed to migrate_raw
are tuples.

This was throwing another exception because of that bug: AttributeError,
saying that tuple has no attribute _type.

dirk-thomas committed:
Merge pull request #917 from efernandez/fix_migrate_raw_exception

Fix BagMigrationException in migrate_raw

dirk-thomas committed:
update changelogs

dirk-thomas committed:
1.12.6

mikepurvis committed:
Fix WallTimerEvent class -> struct

dirk-thomas committed:
Merge pull request #924 from mikepurvis/patch-1

Fix WallTimerEvent class -> struct

mikepurvis committed:
Drop epydoc from rosbag_storage

dirk-thomas committed:
Merge pull request #925 from mikepurvis/patch-2

Drop epydoc from rosbag_storage

Commit:
[rospy] made get_published_topics threadsafe (#958)

* [rospy] made get_published_topics threadsafe

dirk-thomas commented:
Splitting this into two PR: one for Kinetic e.g. moving init code, one for Lunar only with changing order.

mgrrx committed:
Proper logrotation for rosout
